### PR TITLE
Use prebuilt openwhisk images for docker-compose

### PR DESCRIPTION
**Pull request by duynguyen**
Updated the docker-compose tool so that:
- It uses prebuilt images from https://hub.docker.com/u/openwhisk/
- Not building all the whisk images locally anymore
- openwhisk-master is still pulled for building the CLI tool

@ddragosd, @csantanapr : wdyt?

### CODE REVIEW

**Comment by ddragosd** [comment]
@duynguyen great idea ! 👍 
I'm seeing the build passes for `docker-compose` and we could merge it, but I have a couple of questions that I'm adding to the code.

File: docker-compose/Makefile [hunk 1]
```diff
@@ -39,10 +39,10 @@ quick-start-info:
 	echo "$$(tput setaf 2)To invoke the function again use: $$(tput setaf 4)make hello-world$$(tput sgr0)"
 	echo "$$(tput setaf 2)To stop openwhisk use: $$(tput setaf 4)make destroy$$(tput sgr0)"
 
-docker:
-	echo "building the docker images ... "
+init-cli:
```
Comment (ddragosd) [inline docker-compose/Makefile]: Should we preserve `make docker` for local development use-case and add `init-cli` in addition ?

File: docker-compose/docker-compose.yml [hunk 2]
```diff
@@ -51,7 +51,7 @@ services:
 
   # WHISK CONTROLLER
   controller:
-    image: whisk/controller:latest
+    image: openwhisk/controller:6a56c45
```
Comment (ddragosd) [inline docker-compose/docker-compose.yml]: Regarding a static version I see 2 options:
1. keeping a static version ensures `quick-start` runs correctly all the time. The question is how often we upgrade the version.
2. going with `latest` version would keep OW updated all the time, but the `quick-start` might fail when OW configuration updates.

What should we pick here ?

File: docker-compose/Makefile (hunk 1 above)
Comment (duynguyen) [inline docker-compose/Makefile]: +1 yes that makes sense. I will add back the `docker` step for dev use case.

File: docker-compose/docker-compose.yml (hunk 2 above)
Comment (csantanapr) [inline docker-compose/docker-compose.yml]: I pick latest
Can someone help out putting together a jenkinsfile, we can run a jenkins job everytime openwhisk core changes to make sure docker-compose use case still works.
If it fails will send a notification to Slack.

I already have a Jenkins Job to build the images and push them, we just need another one to test docker-compose use case that it's always is working using latest images.

Comment (ddragosd) [inline docker-compose/docker-compose.yml]: > putting together a jenkinsfile, we can run a jenkins job everytime openwhisk core changes

👍  @csantanapr

Comment (duynguyen) [inline docker-compose/docker-compose.yml]: +1 to the `latest` version too.

I can put in a jenkinsfile for verifying docker-compose. Where do you put the jenkins job for building and pushing images?

Comment (csantanapr) [inline docker-compose/docker-compose.yml]: Currently I have a jenkins build here https://builds.apache.org/view/OpenWhisk/job/OpenWhisk-DockerHub/
Using jenkinsfile https://github.com/openwhisk/openwhisk/blob/master/tools/jenkins/apache/dockerhub.groovy

Lets add a jenkinsfile for docker-compose testing here: https://github.com/apache/incubator-openwhisk-devtools/blob/master/tools/jenkins/apache/docker-compose.groovy

I think we can merge this PR without the jenkins integration enable yet and create an issue,  if it works locally, then I can create a new Jenkins Job definition that uses the new jenkins file docker-compose.groovy

Comment (csantanapr) [inline docker-compose/docker-compose.yml]: @duynguyen @ddragosd 
Create issue https://github.com/apache/incubator-openwhisk-devtools/issues/28__

**Comment by ddragosd** [comment]
@duynguyen once https://github.com/apache/incubator-openwhisk-devtools/pull/34 is merged we should be able to use the `latest` tag for the docker images.

**Comment by duynguyen** [comment]
@ddragosd thanks for the fixes in #32 and #34. Now we're able to build successfully with the `latest` tag.

**Comment by ddragosd** [comment]
I've tested this again and everything seems to be working as expected. Unless there are more comments I'm thinking to merge this.

**Comment by ddragosd** [comment]
@duynguyen thanks a lot. I'm merging this now.

**Pull request merged**